include/Dhyana6060Camera.h

```diff
@@ -0,0 +1,257 @@
+    int  getNbHwAcquiredFrames();
+
+    // -- detector info object
+    void getImageType(ImageType& type);
+    void setImageType(ImageType type);
+
+    void getDetectorType(std::string& type);
+    void getDetectorModel(std::string& model);
+    void getDetectorImageSize(Size& size);
+    void getPixelSize(double& sizex, double& sizey);
+
+    // -- Buffer control object
+    HwBufferCtrlObj* getBufferCtrlObj();
+
+    //-- Synch control object
+    void setTrigMode(TrigMode mode);
+    void getTrigMode(TrigMode& mode);
+    bool checkTrigMode(TrigMode mode);
+
+    void setExpTime(double exp_time);
+    void getExpTime(double& exp_time);
+
+    void setLatTime(double lat_time);
+    void getLatTime(double& lat_time);
+
+    void getExposureTimeRange(double& min_expo, double& max_expo) const;
+    void getLatTimeRange(double& min_lat, double& max_lat) const;
+
+    void setNbFrames(int nb_frames);
+    void getNbFrames(int& nb_frames);
+
+    //-- Related to Bin control object
+    void setBin(const Bin& bin);
+    void getBin(Bin& bin);
+    void checkBin(Bin& bin);
+
+    //-- Related to Roi control object
+    void checkRoi(const Roi& set_roi, Roi& hw_roi);
+    void setRoi(const Roi& set_roi);
+    void getRoi(Roi& hw_roi);
+
+    ///////////////////////////////
+    // -- dhyana specific functions
+    ///////////////////////////////
+
+    void getCameraTemperature(double& temp);
+    void setSensorTemperatureTarget(double temp);
+    void getSensorTemperatureTarget(double& temp);
+    void getSensorTemperature(double& temp);
+    void setSensorCoolingType(unsigned type);
+    void getSensorCoolingType(unsigned& type);
+    void setFanSpeed(unsigned speed);
+    void getFanSpeed(unsigned& speed);
+    void setFanType(unsigned type);
+    void getFanType(unsigned& type);
+    void setGlobalGain(unsigned gain);
+    void getGlobalGain(unsigned& gain);
+    void getTucamVersion(std::string& version);
+    void getFirmwareVersion(std::string& version);
+    bool isAcqRunning() const;
+
+    void getFPS(double& fps);	
+    void getTriggerMode(TucamTriggerMode& mode);
+    void setTriggerMode(TucamTriggerMode mode);
+    void getTriggerEdge(TucamTriggerEdge& edge);
+    void setTriggerEdge(TucamTriggerEdge edge);
+    void getOutputSignal(int port, TucamSignal& signal, TucamSignalEdge& edge, int& delay, int& width);
+    void setOutputSignal(int port, TucamSignal signal, TucamSignalEdge edge=kSignalEdgeRising, int delay=-1, int width=-1);
+
+    //TUCAM stuff, use TUCAM notations !
+    TUCAM_INIT          m_itApi; // TUCAM handle Api
+    TUCAM_OPEN          m_opCam; // TUCAM handle camera
+    TUCAM_FRAME         m_frame; // TUCAM frame structure
+    HANDLE              m_hThdEvent; // TUCAM handle event   
+private:
+ //read/copy frame
+    bool readFrame(void *bptr, int& frame_nb);
+    void setStatus(Camera::Status status, bool force);    
+	void _startAcq();
+    inline bool IS_POWER_OF_2(long x)
+    {

```

Comment:
Pour le fun, on peut l'améliorer (thanks to ChatGpt)
inline bool IS_POWER_OF_2(long nombre)
{
return (nombre > 0) && ((nombre & (nombre - 1)) == 0);
}